File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/crypto/GetKeyMetadataTest.java
package org.hyperledger.indy.sdk.crypto;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithSingleWallet;
import org.hyperledger.indy.sdk.wallet.WalletItemNotFoundException;
import org.junit.Before;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertEquals;


public class GetKeyMetadataTest extends IndyIntegrationTestWithSingleWallet {

	private String key;

	@Before
	public void createKey() throws Exception {
		key = Crypto.createKey(wallet, "{}").get();
	}

	@Test
	public void testGetKeyMetadataWorks() throws Exception {
		Crypto.setKeyMetadata(wallet, key, METADATA).get();
		String receivedMetadata = Crypto.getKeyMetadata(wallet, key).get();
		assertEquals(METADATA, receivedMetadata);
	}

	@Test
	public void testGetKeyMetadataWorksForEmptyString() throws Exception {
		Crypto.setKeyMetadata(wallet, key, "").get();
		String receivedMetadata = Crypto.getKeyMetadata(wallet, key).get();
		assertEquals("", receivedMetadata);
	}

	@Test
	public void testGetKeyMetadataWorksForNoKey() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletItemNotFoundException.class));

		Crypto.getKeyMetadata(wallet, VERKEY).get();
	}

	@Test
	public void testGetKeyMetadataWorksForNoMetadata() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletItemNotFoundException.class));

		Crypto.getKeyMetadata(wallet, key).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/CryptoTests/GetKeyMetadataTest.cs
using Hyperledger.Indy.CryptoApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.CryptoTests
{
    [TestClass]
    public class GetKeyMetadataTest : IndyIntegrationTestWithSingleWallet
    {
        private string key;

        [TestInitialize]
        public async Task CreateKey()
        {
            key = await Crypto.CreateKeyAsync(wallet, "{}");
        }

        [TestMethod]
        public async Task TestGetKeyMetadataWorks()
        {
            await Crypto.SetKeyMetadataAsync(wallet, key, METADATA);
            var receivedMetadata = await Crypto.GetKeyMetadataAsync(wallet, key);
            Assert.AreEqual(METADATA, receivedMetadata);
        }

        [TestMethod]
        public async Task TestGetKeyMetadataWorksForEmptyString()
        {
            await Crypto.SetKeyMetadataAsync(wallet, key, string.Empty);
            var receivedMetadata = await Crypto.GetKeyMetadataAsync(wallet, key);
            Assert.AreEqual(string.Empty, receivedMetadata);
        }


        [TestMethod]
        public async Task TestGetKeyMetadataWorksForNoKey()
        {
            await Crypto.SetKeyMetadataAsync(wallet, key, string.Empty);
            var receivedMetadata = await Crypto.GetKeyMetadataAsync(wallet, key);
            Assert.AreEqual(string.Empty, receivedMetadata);
        }

        [TestMethod]
        public async Task TestGetKeyMetadataWorksForNoMetadata()
        {
            var ex = await Assert.ThrowsExceptionAsync<WalletItemNotFoundException>(() =>
               Crypto.GetKeyMetadataAsync(wallet, VERKEY)
           );
        }
    }
}
